File: tinypng/compress.rb
# export TINY_KEY, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
require 'tinify'

Tinify.key = ENV['TINY_KEY']

puts "Correct URL and image"
source = Tinify.from_url("https://s3-eu-west-1.amazonaws.com/tinypng-com-preview/testing/source.png")
puts source.inspect
puts

puts "URL to html page"
begin
  source = Tinify.from_url("http://frankevers.nl")
  puts source.inspect
rescue Tinify::ClientError => e
  puts e
end
puts

puts "Unresolveable http URL"
begin
  source = Tinify.from_url("http://jfadskfjfljkfldas")
  puts source.inspect
rescue Tinify::ClientError => e
  puts e
end
puts

puts "URL wrong protocol"
begin
  source = Tinify.from_url("stuff://jfadskfjfljkfldas")
  puts source.inspect
rescue Tinify::ClientError => e
  puts e
end
puts


# puts "Doing normal compression"
# source.to_file("images/out/small-compressed.jpg")
#
# puts "Resizing image"
# resized = source.resize(
#   method: 'fit',
#   width: 100,
#   height: 80
# )
# resized.to_file("images/out/small-compressed-resized.jpg")
#
# aws_access_key_id = ENV['AWS_ACCESS_KEY_ID']
# aws_secret_access_key = ENV['AWS_SECRET_ACCESS_KEY']
#
# source.store(
#   service: 's3',
#   aws_access_key_id: aws_access_key_id,
#   aws_secret_access_key: aws_secret_access_key,
#   path: 'tinypng-com-preview/testing/small.jpg'
# )
# resized.store(
#   service: 's3',
#   aws_access_key_id: aws_access_key_id,
#   aws_secret_access_key: aws_secret_access_key,
#   path: 'tinypng-com-preview/testing/small-resized.jpg'
# )
#
# puts "\nThank you come again!"
# puts "Compression count: #{Tinify.compression_count}"

File: tinypng/java-compress/src/main/java/nl/frankevers/app/Compress.java
package nl.fevers.app;
import com.tinify.*;
import java.io.IOException;

public class Compress {
    public static void main(String[] args) throws java.io.IOException {
        Tinify.setKey(System.getenv("TINY_KEY"));
        System.out.println("Key: " + System.getenv("TINY_KEY"));
        System.out.println("Compression-Count: " + Tinify.compressionCount());

        // File
        Tinify.fromFile("../images/in/small.png").toFile("../images/out/small-compressed-url.png");
        System.out.println("Compression done. Compression-Count: " + Tinify.compressionCount());

        // URL
        Source tiny = Tinify.fromUrl("https://raw.githubusercontent.com/tinify/tinify-ruby/master/test/examples/voormedia.png");
        tiny.toFile("../images/out/small-compressed-url.png");
        System.out.println("Compression done. Compression-Count: " + Tinify.compressionCount());
    }
}